[CI] RandomizedTimeSeriesIT testRateGroupByNothing failing

**Build Scans:**
- [elasticsearch-periodic #12617 / release-tests](https://gradle-enterprise.elastic.co/s/h3gsuffpxoppu)
- [elasticsearch-periodic #12617 / release-tests](https://gradle-enterprise.elastic.co/s/yzvupy6lbzbhe)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT.testRateGroupByNothing" -Dtests.seed=C6AEF468C1777AC4 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=hsb-DE -Dtests.timezone=America/Argentina/San_Luis -Druntime.java=26
```

**Applicable branches:**
9.5

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRateGroupByNothing'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failed. Agg: RATE | Failed windows: 1 windows(1.6666666666666667%):
Failed for row:
[13, 17.910770829114945, 8.737311547010597, 0.9566061263070169, 2025-07-31T09:46:00.000Z]
Wanted: RateStats[count=13, max=RateRange[lower=17.89286004166554, upper=17.928681583290494], avg=RateRange[lower=8.838080002509509, upper=8.855773856368385], min=RateRange[lower=0.9556495216969361, upper=0.9575627339525855], sum=RateRange[lower=114.89504003262361, upper=115.12506013278902]]
Exception: 
Expected: (a value less than or equal to <8.855773856368385> and not a value less than <8.838080002509509>)
     but: not a value less than <8.838080002509509> was <8.737311547010597>
```

**Issue Reasons:**
- [9.5] 2 consecutive failures in test testRateGroupByNothing
- [9.5] 2 consecutive failures in step release-tests
- [9.5] 1 consecutive failures in pipeline elasticsearch-periodic
- [9.5] 2 failures in test testRateGroupByNothing (4.8% fail rate in 42 executions)
- [9.5] 2 failures in step release-tests (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-core-infra.

:eyes:

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -766,9 +766,6 @@ tests:
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/155532
-- class: org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT
-  method: testRateGroupByNothing
-  issue: https://github.com/elastic/elasticsearch/issues/155684
 - class: org.elasticsearch.packaging.test.PackageTests
   method: test70RestartServer
   issue: https://github.com/elastic/elasticsearch/issues/155704

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
@@ -503,24 +503,31 @@ public class RandomizedTimeSeriesIT extends AbstractEsqlIntegTestCase {
         assert values.size() >= 2 : "At least two points are required for extrapolation";
 
         double increase = computeCounterIncreaseBetween(values, temporality);
-        long firstTs = values.getFirst().timestamp().toEpochMilli();
-        long lastTs = values.getLast().timestamp().toEpochMilli();
 
-        final long sampleTs = lastTs - firstTs;
-        final double averageSampleInterval = sampleTs * 1.0 / values.size();
-        final double slope = increase / sampleTs;
+        // Use seconds-based double arithmetic to match the production code in
+        // RateDoubleGroupingAggregatorFunction#extrapolateToBoundary, where timestamps are converted
+        // to seconds via epoch_ms / 1000.0. This matters for the gap clamping comparison
+        // (gap > averageSampleInterval * 1.1) which can evaluate differently under double precision.
+        long firstTsMs = values.getFirst().timestamp().toEpochMilli();
+        long lastTsMs = values.getLast().timestamp().toEpochMilli();
+        assert firstTsMs != lastTsMs;
 
-        assert firstTs != lastTs;
+        double firstTsSec = firstTsMs / 1000.0;
+        double lastTsSec = lastTsMs / 1000.0;
+        final double sampleTsSec = lastTsSec - firstTsSec;
+        final double averageSampleInterval = sampleTsSec / values.size();
+        final double slope = increase / sampleTsSec;
 
         long millisInWindow = secondsInWindow * 1000L;
-        long tbucketStart = firstTs / millisInWindow * millisInWindow;
-        long tbucketEnd = tbucketStart + millisInWindow;
+        long tbucketStartMs = firstTsMs / millisInWindow * millisInWindow;
+        double tbucketStartSec = tbucketStartMs / 1000.0;
+        double tbucketEndSec = (tbucketStartMs + millisInWindow) / 1000.0;
 
         double gap;
         if (isLowerBoundary) {
-            gap = firstTs - tbucketStart;
+            gap = firstTsSec - tbucketStartSec;
         } else {
-            gap = tbucketEnd - lastTs;
+            gap = tbucketEndSec - lastTsSec;
         }
         if (gap > 0) {
             if (gap > averageSampleInterval * 1.1) {

